File: examples/p2_fun_circles.py
import g2d
from random import randrange

def tick():
    if g2d.key_pressed("LeftButton"):
        x, y = g2d.mouse_position()
        g2d.set_color((randrange(256), randrange(256), randrange(256)))
        if x <= 25 and y <= 25 and g2d.confirm("Exit?"):
            g2d.close_canvas()
        else:
            g2d.fill_circle((x, y), 25)

g2d.init_canvas((480, 360))
g2d.main_loop(tick)

File: go/p2_fun_circles.go
package main

import . "g2d"

func tick() {
    if MouseClicked() {
        pos := MousePosition()
        SetColor(Color{RandInt(0, 255), RandInt(0, 255), RandInt(0, 255)})
        if pos.X <= 25 && pos.Y <= 25 && Confirm("Exit?") {
            CloseCanvas()
        } else {
            FillCircle(pos, 25)
        }
    }
}

func main() {
    InitCanvas(Point{480, 360})
    MainLoop(tick)  // Call tick 30 times/second
}
